src/lib/react/VideoView.tsx

```diff
@@ -7,28 +7,49 @@ interface VideoViewSelfProps {
     mirror?: boolean;
     style?: React.CSSProperties;
     onResize?: ({ width, height, stream }: { width: number; height: number; stream: MediaStream }) => void;
+    onSetAspectRatio?: ({ aspectRatio }: { aspectRatio: number }) => void;
 }
 
 type VideoViewProps = VideoViewSelfProps &
     React.DetailedHTMLProps<React.VideoHTMLAttributes<HTMLVideoElement>, HTMLVideoElement>;
 
-export default ({ muted, mirror = false, stream, onResize, ...rest }: VideoViewProps) => {
-    const videoEl = useRef<HTMLVideoElement>(null);
+export default ({ muted, mirror = false, stream, onResize, onSetAspectRatio, ...rest }: VideoViewProps) => {
+    const videoEl = React.useRef<HTMLVideoElement>(null);
 
-    useEffect(() => {
-        if (!videoEl.current || !onResize) {
+    React.useEffect(() => {
+        if (!videoEl.current) return;
+        // Send aspect ratio on first render
+        if (onSetAspectRatio) {
+            const h = videoEl.current.videoHeight;
+            const w = videoEl.current.videoWidth;
+            if (w && h && w + h > 20) {
```

Comment:
Maybe extract this or document it a bit? I see this is used below as well

Reply:
I just removed this for now.